ACG/AutoRouterShield.py

```diff
@@ -0,0 +1,714 @@
+
+        dirs = []
+
+        for i in range(len(manh) - 1):
+            pt0 = manh[i]
+            pt1 = manh[i + 1]
+
+            if pt0[0][0] > pt1[0][0]:
+                dirs.append('-x')
+            elif pt0[0][0] < pt1[0][0]:
+                dirs.append('+x')
+            elif pt0[0][1] > pt1[0][1]:
+                dirs.append('-y')
+            else:
+                dirs.append('+y')
+
+        router1 = None
+        router2 = None
+
+        for i in range(len(dirs)):
+            if i == 0:
+                if start_pt[0] > manh[1][0][0]:
+                    start = (0, -1)
+                elif start_pt[0] < manh[1][0][0]:
+                    start = (0, 1)
+                elif start_pt[1] > manh[1][0][1]:
+                    start = (1, 0)
+                else:
+                    start = (-1, 0)
+                shield1_start = ((start_pt[0] + start[0] * parallel_spacing / 2,
+                                  start_pt[1] + start[1] * parallel_spacing / 2), start_met)
+                shield2_start = ((start_pt[0] - start[0] * parallel_spacing / 2,
+                                  start_pt[1] - start[1] * parallel_spacing / 2), start_met)
+                router1 = EZRouterShield(self.gen)
+                router2 = EZRouterShield(self.gen)
+                router1.new_route_from_location(shield1_start[0], start_dir, start_met, 0.5)
+                router2.new_route_from_location(shield2_start[0], start_dir, start_met, 0.5)
+            else:
+                pt0 = manh[i]
+
+                direc = self.shield_dict[dirs[i - 1]][dirs[i]]
+                router1.add_route_points(
+                    [(pt0[0][0] + direc[0] * parallel_spacing / 2,
+                      pt0[0][1] + direc[1] * parallel_spacing / 2)], pt0[1],
+                    width=self.route_point_dict[pt0[0]])
+                router2.add_route_points(
+                    [(pt0[0][0] - direc[0] * parallel_spacing / 2,
+                      pt0[0][1] - direc[1] * parallel_spacing / 2)], pt0[1],
+                    width=self.route_point_dict[pt0[0]])
+
+        if manh[-2][0][0] > manh[-1][0][0]:
+            end = (0, -1)
+        elif manh[-2][0][0] < manh[-1][0][0]:
+            end = (0, 1)
+        elif manh[-2][0][1] > manh[-1][0][1]:
+            end = (1, 0)
+        else:
+            end = (-1, 0)
+
+        router1.add_route_points(
+            [(manh[-1][0][0] + end[0] * parallel_spacing / 2,
+              manh[-1][0][1] + end[1] * parallel_spacing / 2)], manh[-1][1],
+            width=self.route_point_dict[manh[-1][0]])
+        router2.add_route_points(
+            [(manh[-1][0][0] - end[0] * parallel_spacing / 2,
+              manh[-1][0][1] - end[1] * parallel_spacing / 2)], manh[-1][1],
+            width=self.route_point_dict[manh[-1][0]])
+
+        router1.cardinal_router_variant()
+        router2.cardinal_router_variant()
+
+        return self
+
+    @staticmethod
+    def process_manh(manh):
```

Comment:
In what cases does this function actually remove points from the manhattanized list? Can we just incorporate this directly into the base `manhattanize_point_list()` method?